go/logger/global.go

```diff
@@ -12,7 +15,21 @@ var globalLock sync.Mutex
 var stderrIsTerminal = isatty.IsTerminal(os.Stderr.Fd())
 var currentLogFileWriter *LogFileWriter
 
+func getBufferedErrorWriter() io.Writer {
+	writer := ErrorWriter()
+	buf := bufio.NewWriter(writer)
+	go func() {
+		for range time.Tick(300 * time.Millisecond) {
+			// TODO: do we care if this errors
+			buf.Flush()
+		}
+	}()
```

Comment:
This goroutine will leak. You'll need to store a ticker and `Stop()` it when this writer needs to be shutdown.

Reply:
I agree that using `time.NewTicker()` is a better idea than `time.Tick`.

Reply:
Also a good idea to call `Flush()` when shutdown happens.